src/phase-manager.ts

```diff
@@ -628,6 +632,44 @@ export class PhaseManager {
     this.createAndUnshiftPhase("FaintPhase", battlerIndex, preventEndure, destinyTag, grudgeTag, source);
   }
 
+  /**
+   * Unshifts a sequence of phases to switch out a Pokemon on the field
+   * @param battlerIndex - The {@linkcode BattlerIndex} of the Pokemon to switch out
+   * @param switchType - (Default {@linkcode SwitchType.SWITCH}) The {@linkcode SwitchType | type} of switch to apply
+   * @param switchInIndex - (Default `-1`) The index of the party Pokemon to switch into
+   * the target Pokemon's place. If set to `-1`, the Pokemon to switch in is instead resolved
+   * during the {@linkcode SwitchPhase}.
+   */
+  public queueBattlerSwitchOut(
+    battlerIndex: BattlerIndex,
+    { switchType = SwitchType.SWITCH, switchInIndex = -1, when = "eager", phaseKey }: BattlerSwitchOutInit = {},
+  ): void {
+    const phases = [
+      this.createPhase("RecallPhase", battlerIndex, switchType),
+      this.createPhase("SwitchPhase", battlerIndex, switchType, switchInIndex),
+    ] as const;
+
+    const validatePhaseId = () => {
+      if (!phaseKey) {
+        throw new Error("`phaseId` is required if `when` is 'before' or 'after'");
+      }
+    };
+
+    switch (when) {
+      case "eager":
+        this.unshiftPhase(...phases);
+        break;
+      case "before":
+        validatePhaseId();
+        this.prependToPhase(phaseKey!, ...phases);
+        break;
+      case "after":
+        validatePhaseId();
+        this.prependToPhase(phaseKey!, ...phases);
```

Comment:
both "before" and "after" use `prependToPhase`, should it be this?
```suggestion
        this.appendToPhase(phaseKey!, ...phases);
```